libs/blocks/comparison-table/comparison-table.js

```diff
@@ -0,0 +1,498 @@
+import { createTag, getConfig } from '../../utils/utils.js';
+import { decorateButtons } from '../../utils/decorate.js';
+import { replaceKeyArray } from '../../features/placeholders.js';
+import { getMetadata } from '../section-metadata/section-metadata.js';
+
+const COLUMN_TYPES = { PRIMARY: 'primary' };
+
+function equalHeight(el) {
+  const calculateMaxHeight = (elements) => Math.max(...elements.map((p) => {
+    const styles = window.getComputedStyle(p);
+    return p.offsetHeight
+      - parseFloat(styles.paddingTop) - parseFloat(styles.paddingBottom)
+      - parseFloat(styles.borderTopWidth) - parseFloat(styles.borderBottomWidth);
+  }));
+
+  const setupHeightHandler = (handler) => {
+    const resizeObserver = new ResizeObserver((entries) => {
+      if (entries.some((entry) => entry.contentBoxSize || entry.borderBoxSize)) handler();
+    });
+    el.querySelectorAll('.table-cell div, .sub-header-item-container')
+      .forEach((element) => resizeObserver.observe(element));
+    return resizeObserver;
+  };
+
+  const performEqualHeightForElements = (parentSelector, childSelector, targetSelector) => {
+    el.querySelectorAll(parentSelector).forEach((parent) => {
+      const children = parent.querySelectorAll(childSelector);
+      if (!children.length) return;
+      const elementsByPosition = [];
+
+      children.forEach((child) => {
+        const targets = child.querySelectorAll(targetSelector);
+        targets.forEach((target, index) => {
+          if (!elementsByPosition[index]) elementsByPosition[index] = [];
+          target.style.minHeight = 'auto';
+          elementsByPosition[index].push(target);
+        });
+      });
+
+      elementsByPosition.forEach((elements) => {
+        if (!elements.length) return;
+        const maxHeight = calculateMaxHeight(elements);
+        elements.forEach((element) => {
+          if (maxHeight === 0) element.classList.add('zero-height');
+          element.style.minHeight = `${maxHeight}px`;
+        });
+      });
+    });
+  };
+
+  const configs = [
+    ['.header-content-wrapper', '.header-item', '.sub-header-item-container:not(:last-of-type)'],
+    ['.table-row', '.table-cell', 'div'],
+    ['.header-content-wrapper', '.header-item', '.description'],
+  ];
+
+  const observers = configs.map(([parent, child, target]) => setupHeightHandler(
+    () => performEqualHeightForElements(parent, child, target),
+  ));
+
+  return () => observers.forEach((observer) => observer?.disconnect());
+}
+
+const getFirstVisibleColumnIndex = (el) => {
+  const firstVisible = [...el.querySelectorAll('.header-item[data-column-index]')]
+    .find((item) => !item.classList.contains('hidden'));
+  return firstVisible ? +firstVisible.getAttribute('data-column-index') : -1;
+};
+
+function syncAccessibilityHeaders(el) {
+  const accessibilityHeaderRow = el.querySelector('.accessibility-header-row');
```

Comment:
probably also makes sense to have some defensive code here in case these elements don't exist to avoid a null pointer exception